---
title: "Getting Started" 
author: "<h4>Authors: <i>`r auths <- eval(parse(text = gsub('person','c',read.dcf('../DESCRIPTION', fields = 'Authors@R'))));paste(auths[names(auths)=='given'],auths[names(auths)=='family'], collapse = ', ')`</i></h4>" 
date: "<h4>Vignette updated: <i>`r format( Sys.Date(), '%b-%d-%Y')`</i></h4>"
output:
  html_document
vignette: >
    %\VignetteIndexEntry{Getting Started} 
    %\usepackage[utf8]{inputenc}
    %\VignetteEngine{knitr::rmarkdown}
---


```{r}
library(ThreeWayTest)
library(MSKAT)
library(AssocTests)
```

# Simulation

## Load data 

```{r setup, include=TRUE} 
data("data_matrix_final") 
data("covariance_matrix_data")
data("gene_list")
data("gene_length_list") 
PCA_data <- ThreeWayTest::snpinfo_12
final_1kg_genotype_correct <- ThreeWayTest::final_1kg_genotype_correct_sub
```


```{r} 
used_PCA_SNP<-intersect(PCA_data$V1, final_1kg_genotype_correct$SNP)
PCA_data_1000_genome_index<-match(used_PCA_SNP,final_1kg_genotype_correct$SNP)
PCA_data_1000_genome<-final_1kg_genotype_correct[PCA_data_1000_genome_index,]
PCA_data_clean<-PCA_data_1000_genome[,7:min(2510,ncol(PCA_data_1000_genome))]

genoFile <- tempfile(fileext = "pcocG.eg.txt")
write.table(PCA_data_clean, 
            file = genoFile, 
            quote = FALSE,
            sep = "", 
            row.names = FALSE, 
            col.names = FALSE)
outFile.txt <- tempfile(fileext = "pcoc.result.txt")
PCA_result<-AssocTests::pcoc(genoFile = genoFile,
                             outFile.txt = outFile.txt,
                             n.MonteCarlo = 1000,
                             num.splits = 10,
                             miss.val = 9)
covariates<-PCA_result$principal.coordinates[,1:2]

```


```{r}

n_sim=1000000
q=6
gamma_value<-1
phenotype_covariance<-stats::cor(covariance_matrix_data[8:13])
select_gene<-gene_list[9592]
match_char=paste(select_gene,"\\(",sep = "")
match_line=which(grepl(match_char,data_matrix_final$ANNOT))
genotype_data<-t(final_1kg_genotype_correct[match_line,7:min(2510,ncol(final_1kg_genotype_correct))])
adj_geno<-stats::lm(genotype_data~covariates)$residuals
genotype_covariance<-stats::cor(adj_geno)
n_sample<-nrow(genotype_data)
geno_length<-ncol(genotype_data)
cutoff_value<-seq(from=0.1,to=1,by=0.1)
c<-as.matrix(rnorm(2504,0,1))
gamma_vec<-as.matrix(rep(gamma_value,q))
c_mat<-c%*%t(gamma_vec)
```

```{r}
pheno_rho<-0.8
covariance_matrix<- ThreeWayTest::true_cov_mat_generation_autoregressive(
    m = q, 
    rho = pheno_rho)
covariance_matrix<-stats::cor(covariance_matrix_data[8:13])

B_mat<-matrix(0, nrow = geno_length, ncol = q)

p_value_MGAS<-rep(NA,n_sim)
p_value_TWT<-rep(NA,n_sim)
p_value_metaCCA<-rep(NA,n_sim)
p_value_MAT<-rep(NA,n_sim)
p_1<-rep(NA,n_sim)
p_2<-rep(NA,n_sim)
p_3<-rep(NA,n_sim)

```

```{r}

est_pheno_rho_mat<-covariance_matrix
total_mat<-methods::kronecker(genotype_covariance,est_pheno_rho_mat)
null_distribution_T3<-ThreeWayTest::generate_null_distribution_T3(
    m = q*geno_length,
    n = 10000,
    correlation_matrix = total_mat,
    cutoff_value = cutoff_value)
coefficient_matrix_T3<-approximate_distribution_coefficient_estimate_T3(
    null_distribution_matrix = null_distribution_T3)
for (i in 1:n_sim) {
  eplison<- MASS::mvrnorm(n=n_sample,rep(0,q),covariance_matrix)
  Y_mat<-genotype_data%*%B_mat+c_mat+eplison
  #for (j in 1:ncol(independent_snp_mat)) {
  #  for (k in 1:q) {
  #    model<-lm(Y_mat[,k]~independent_snp_mat[,j])
  #    wald <- coef(model)[2] / sqrt((diag(vcov(model)))[2])
  #    wald_mat_independent[j,k]<-wald
  #  }
  #}
  #est_pheno_rho_mat<-cor(wald_mat_independent)
  Z_mat<-matrix(NA,ncol(genotype_data),ncol(Y_mat))
  std_beta_mat<-matrix(NA,ncol(genotype_data),ncol(Y_mat))
  for (j in 1:ncol(genotype_data)) {
    for (k in 1:ncol(Y_mat)) {
      model<- stats::lm(Y_mat[,k]~genotype_data[,j]+covariates+c)
      wald <- stats::coef(model)[2] / sqrt(diag(vcov(model)))[2]
      Z_mat[j,k] <- wald
      std_beta_mat[j,k]<- coef(model)[2]/sqrt(n_sample*(diag(vcov(model))[2]))
    }
  }
  z_vector<-as.vector(t(Z_mat))
  p_value_MGAS[i]<-ThreeWayTest::MGAS(z_vector,genotype_covariance,est_pheno_rho_mat)
  cc<-ThreeWayTest::new_method(Z_mat,genotype_covariance,est_pheno_rho_mat,cutoff_value,coefficient_matrix_T3)
  p_value_TWT[i]<-cc$p_value_final
  p_1[i]<-cc$p_1
  p_2[i]<-cc$p_2
  p_3[i]<-cc$p_3
  p_value_metaCCA[i]<-ThreeWayTest::metaCCA(genotype_covariance,est_pheno_rho_mat,std_beta_mat,n_sample)
  p_value_MAT[i]<-MSKAT::MSATS(Z_mat,est_pheno_rho_mat,genotype_covariance)$p.value[1]
  print(i)
}
```
 

```{r}
sum(as.numeric(p_value_MGAS<0.00001),na.rm = TRUE)
sum(as.numeric(p_value_TWT<0.00001),na.rm = TRUE)
sum(as.numeric(p_value_metaCCA<0.00001),na.rm = TRUE)
sum(as.numeric(p_value_MAT<0.00001),na.rm = TRUE)
sum(as.numeric(p_1<0.00001),na.rm = TRUE)
sum(as.numeric(p_2<0.00001),na.rm = TRUE)
sum(as.numeric(p_3<0.00001),na.rm = TRUE)

sum(as.numeric(p_value_MGAS<0.0001),na.rm = TRUE)
sum(as.numeric(p_value_TWT<0.0001),na.rm = TRUE)
sum(as.numeric(p_value_metaCCA<0.0001),na.rm = TRUE)
sum(as.numeric(p_value_MAT<0.0001),na.rm = TRUE)
sum(as.numeric(p_1<0.0001),na.rm = TRUE)
sum(as.numeric(p_2<0.0001),na.rm = TRUE)
sum(as.numeric(p_3<0.0001),na.rm = TRUE)

sum(as.numeric(p_value_MGAS<0.001),na.rm = TRUE)
sum(as.numeric(p_value_TWT<0.001),na.rm = TRUE)
sum(as.numeric(p_value_metaCCA<0.001),na.rm = TRUE)
sum(as.numeric(p_value_MAT<0.001),na.rm = TRUE)
sum(as.numeric(p_1<0.001),na.rm = TRUE)
sum(as.numeric(p_2<0.001),na.rm = TRUE)
sum(as.numeric(p_3<0.001),na.rm = TRUE)

sum(as.numeric(p_value_MGAS<0.01),na.rm = TRUE)
sum(as.numeric(p_value_TWT<0.01),na.rm = TRUE)
sum(as.numeric(p_value_metaCCA<0.01),na.rm = TRUE)
sum(as.numeric(p_value_MAT<0.01),na.rm = TRUE)
sum(as.numeric(p_1<0.01),na.rm = TRUE)
sum(as.numeric(p_2<0.01),na.rm = TRUE)
sum(as.numeric(p_3<0.01),na.rm = TRUE)
```



# Session Info 

<details> 

```{r Session Info}
utils::sessionInfo()
```

</details>  

<br>
